wagtail_localize_rws_languagecloud/templates/wagtailadmin/shared/locale_selector.html

```diff
@@ -0,0 +1,33 @@
+{# Based on listing/_button_with_dropdown.html #}
+{% load wagtailadmin_tags wagtaillocalizerwslanguagecloud_tags %}
+{% if translations %}
+    {% language_cloud_statuses as lc_statuses %}
+    <div class="c-dropdown t-inverted {{ class }}" style="display: inline-block;" data-dropdown>
+        <a href="javascript:void(0)" aria-label="{{ locale.get_display_name }}" class="c-dropdown__button  u-btn-current">
+            {% icon name="site" class_name="default" %}
+            {{ locale.get_display_name }}
+            <div data-dropdown-toggle class="o-icon c-dropdown__toggle c-dropdown__togle--icon [ icon icon-arrow-down ]">
+                {% icon name="arrow-down" %}{% icon name="arrow-up" %}
+            </div>
+        </a>
+        <div class="t-dark">
+            <ul class="c-dropdown__menu u-toggle  u-arrow u-arrow--tl u-background">
+            {% for translation in translations %}
+                <li class="c-dropdown__item " style="float: none; margin: 0.5em;">
+                    <a href="{{ translation.url }}" aria-label="{{ translation.locale.get_display_name }}" class="u-link is-live">
+                      {{ translation.locale.get_display_name }}
+                      {% for status_locale, status in lc_statuses %}
```

Comment:
This code isn't particularly efficient - we have to iterate over `lc_statuses` for each translation so it gets worse as we add more languages.

If we returned a dict from `language_cloud_statuses` instead of an array, we could reduce this to a hash lookup by using another template tag (basically this trick https://stackoverflow.com/a/10700142 but we'd need to account for the `KeyError` case)

I think this would also make the intent of the code clearer.

Reply:
Do you think this is a huge issue? I could make it more efficient, but the reason I didn't was that the performance gains will likely be imperceptible.

I do agree it's not as readable as it could be, but I think adding a custom template tag to allow dictionary key access won't be much better, given that it's not a standard DJT feature.

I think explicitly writing out the logic in the template is the most straightforward way to do this if a little verbose.

Would you be happy if I just wrote down the intent as a comment to add more clarity?

Let me know if you feel strongly about the tag returning a dictionary and I will change it.

Reply:
Lets optimise for readability over performance. Happy to defer to your opinion on this :+1: